src/fs_io.cc

```diff
@@ -37,6 +37,25 @@ static void cxx_xfree(void *ptr)
     xfree(ptr);
 }
 
+dwrite_q::dwrite_q(const size_t aSize, char *aBuffer, FREE *aFree) :
+    buf(aBuffer),
+    capacity(aSize),
+    free_func(aFree)
+{
+    if (!buf) {
```

Comment:
```suggestion
    assert(buf || !free_func);
    if (!buf) {
```

Ideally, this `if` statement should not exist at all -- this constructor should only be used for cases where buffer is known. Implementing that ideal correctly (e.g., without code duplication) requires additional work. I do not insist on reaching that ideal.

The trivial change requested above is the second-best alternative that just clarifies that we expect callers to keep aBuffer and aFree parameters in sync. For example, callers that do not supply a buffer must not supply a buffer freeing function either.

Reply:
This change request was resolved without any discussion or code changes. Please do not resolve this change request without applying the suggested change (or otherwise explicitly addressing it).